packages/framework/tests/Unit/RouteKeyTest.php

```diff
@@ -93,6 +95,56 @@ public function testFromPageWithCustomNestedOutputDirectory()
         $this->assertEquals(new RouteKey('foo/bar/baz'), RouteKey::fromPage(MarkdownPage::class, 'baz'));
     }
 
+    public function testItExtractsCoreIdentifierPartFromNumericalFilenamePrefix()
+    {
+        $this->assertSame('test', RouteKey::fromPage(InMemoryPage::class, '01-test')->get());
+    }
+
+    public function testItExtractsCoreIdentifierPartFromNumericalFilenamePrefixWithKebabCaseSyntax()
+    {
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '01-foo')->get());
+        $this->assertSame('bar', RouteKey::fromPage(InMemoryPage::class, '02-bar')->get());
+        $this->assertSame('baz', RouteKey::fromPage(InMemoryPage::class, '03-baz')->get());
+    }
+
+    public function testItExtractsCoreIdentifierPartFromNumericalFilenamePrefixWithSnakeCaseSyntax()
+    {
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '01_foo')->get());
+        $this->assertSame('bar', RouteKey::fromPage(InMemoryPage::class, '02_bar')->get());
+        $this->assertSame('baz', RouteKey::fromPage(InMemoryPage::class, '03_baz')->get());
+    }
+
+    public function testItExtractsCoreIdentifierPartFromNumericalFilenamePrefixRegardlessOfLeadingZeroes()
+    {
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '123-foo')->get());
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '0123-foo')->get());
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '00123-foo')->get());
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '000123-foo')->get());
+        $this->assertSame('foo', RouteKey::fromPage(InMemoryPage::class, '0000123-foo')->get());
+    }
+
+    public function testItExtractsCoreIdentifierPartFromNumericalFilenamePrefixForNestedIdentifiers()
+    {
+        $this->assertSame('foo/bar', RouteKey::fromPage(InMemoryPage::class, 'foo/01-bar')->get());
+        $this->assertSame('foo/bar/baz', RouteKey::fromPage(InMemoryPage::class, 'foo/bar/02-baz')->get());
+        $this->assertSame('foo/01-bar/baz', RouteKey::fromPage(InMemoryPage::class, 'foo/01-bar/03-baz')->get());
```

Comment:
We need to see how we should handle this. Presumably we would want to remove the prefix from subdirectories too, since we have documented that it works for subdirectories (which still needs to be implemented). This could get complex...

Reply:
Resolved in 0bcb133786efff896138ea203188c17caedbabc3